File: 프로그래머스/lv2/김미래/124 나라의 숫자.cpp
#include <string>
#include <vector>
#include <iostream>
#include <string>
using namespace std;

string solution(int n) {
    string answer = "";
    
    while(n>0){
        
        int remain = n % 3;
        n = n / 3 ;
        
        if(remain == 0){
            answer = "4" + answer;
            n--;
            
        }else{
            answer = to_string(remain) + answer;
        }
        
    }
    
    return answer;
}

File: 프로그래머스/lv2/이선영/124 나라의 숫자.java
import java.util.*;

class Solution {
    public String solution(int n) {
        String answer = "";
        StringBuilder sb = new StringBuilder();
        
        while(n>0){
            int num = n%3;
            if(num == 0){ // 4
                sb.insert(0,'4');
                n/=3;
                n--;
            }else{ // 1,2
                sb.insert(0,Integer.toString(num));
                 n /=3;
            }
        }
        
        answer = sb.toString();
        return answer;
    }
}
